scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -2218,12 +2298,114 @@ static bool layered_monitor(void)
 	return true;
 }
 
+u64 antistall_set(u64 dsq_id, u64 now_sec) 
+{
+	struct task_struct *p;
+	struct task_ctx *tctx;
+	s32 cpu;
+	u64 *antistall_dsq, *delay, cur_delay, empty, runnable_at_sec;
+	bool first_pass;
+	u32 zero;
+	
+	zero = 0;
+	empty = -1;
+
+	if (!dsq_id || !now_sec)
+		return 0;
+	bpf_rcu_read_lock();	
+	bpf_for_each(scx_dsq, p, dsq_id, 0) {
+		
+		if (!(tctx = lookup_task_ctx(p)))
+			goto unlock;
+		
+		runnable_at_sec = p->scx.runnable_at / CONFIG_HZ;
+		cur_delay = now_sec - runnable_at_sec;
+		if (cur_delay <= antistall_sec || cur_delay > 100)
```

Comment:
Why does it jump to unlock if `cur_delay > 100`?

Reply:
Sometimes delay numbers make no sense (i.e. I've seen it go as high as 2k IIRC). I figured that happens because this is inherently racy maybe. Will post logs of this in a bit.

Reply:
So, jiffies wrap and I don't think you can convert it to secs and then calculate delta. You should first calculate delta and then divide so that wrapping doesn't affect the result.

Reply:
Something like:
```
now = bpf_jiffies64();
runable_at = READ_ONCE(p->scx.runnable_at);

if (vtime_after(now, runnable_at)) // note that runnable_at can be in the future
        delta_secs = (now - runnable_at) / CONFIG_HZ;
else
        delta_secs = 0;
```

Reply:
Thanks, that did it, no weird delay numbers anymore.